File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFreezeTicks.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPFreezeTicks implements PlayerStateProperty<Integer> {
    @Override
    public String getConfigName() {
        return "freeze_ticks";
    }

    @Override
    public Integer getDefaultSerializedProperty() {
        return 0;
    }

    @Override
    public Integer getSerializedPropertyFromPlayer(Player player) {
        return player.getFreezeTicks();
    }

    @Override
    public void restoreProperty(Player player, Integer ticks) {
        player.setFreezeTicks(ticks);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPFreezeTicks.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPFreezeTicks extends PlayerStateProperty[Integer]:
  override def getConfigName: String = "freeze_ticks"

  override def getDefaultSerializedProperty: Integer = 0

  override def getSerializedPropertyFromPlayer(player: Player): Integer =
    player.getFreezeTicks

  override def restoreProperty(player: Player, ticks: Integer): Unit =
    player.setFreezeTicks(ticks)